### PR TITLE
[Uid] Document the methods to generate and inspect UIDs

### LINKED ISSUE #15032: [Uid] Add Generate and Inspect commands
| Q            | A
| ------------ | ---
| Feature PR   | symfony/symfony#39883
| PR author(s) | @fancyweb
| Merged in    | 5.3-dev

### PR DESCRIPTION
**Pull request by javiereguiluz**
Fixes #15032.

### CODE REVIEW

File: components/uid.rst [hunk 1]
```diff
@@ -370,6 +370,121 @@ of the ULID parameters::
...
+     ---------------------- --------------------------------------
+      Version                4
+      Canonical (RFC 4122)   d0a3a023-f515-4fe0-915c-575e63693998
+      Base 58                SmHvuofV4GCF7QW543rDD9
+      Base 32                6GMEG27X8N9ZG92Q2QBSHPJECR
+     ---------------------- --------------------------------------
+
+.. versionadded:: 5.3
+
+    The commands to inspect and generate UUIDs/ULIDs were introduced in Symfony 5.3.
```
Comment (wouterj) [inline components/uid.rst]: Don't we usually put these *before* the documentation of the feature nowadays?

File: components/uid.rst [hunk 2]
```diff
@@ -370,6 +370,121 @@ of the ULID parameters::
...
+        use Symfony\Component\Uid\Command\GenerateUlidCommand;
+        use Symfony\Component\Uid\Command\GenerateUuidCommand;
+        use Symfony\Component\Uid\Command\InspectUlidCommand;
+        use Symfony\Component\Uid\Command\InspectUuidCommand;
+
+        return static function (ContainerConfigurator $configurator): void {
+            $services = $configurator->services()
+                ->defaults()
+                ->autowire()
+                ->autoconfigure();
```
Comment (wouterj) [inline components/uid.rst]:
```suggestion
            // ...
```

File: components/uid.rst [hunk 3]
```diff
@@ -370,6 +370,121 @@ of the ULID parameters::
...
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # services.yaml
+        services:
+            Symfony\Component\Uid\Command\GenerateUlidCommand: ~
+            Symfony\Component\Uid\Command\GenerateUuidCommand: ~
+            Symfony\Component\Uid\Command\InspectUlidCommand: ~
+            Symfony\Component\Uid\Command\InspectUuidCommand: ~
```
Comment (wouterj) [inline components/uid.rst:393]: I understand why they are optional, but this seems like a really non-DX friendly feature this way. @fancyweb @nicolas-grekas is there a particular reason not to add a config option for this?

**Review by wouterj — APPROVED** [review]

File: components/uid.rst (hunk 1 above)
Comment (javiereguiluz) [reply components/uid.rst]: Yes. Done!

File: components/uid.rst [hunk 4]
```diff
@@ -370,6 +370,118 @@ of the ULID parameters::
...
+
+This component provides several commands to generate and inspect UUIDs/ULIDs in
+the console. They are not enabled by default, so you must add the following
+configuration in your application before using these commands:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # services.yaml
```
Comment (OskarStark) [inline components/uid.rst]:
```suggestion
        # config/services.yaml
```

same for the `services.php` path?

Comment (javiereguiluz) [reply components/uid.rst]: Good catch! Fixed. Thanks!

**Review by OskarStark — APPROVED** [review]

File: components/uid.rst (hunk 3 above)
Comment (94noni) [reply components/uid.rst:393]: can those commands be enabled by default but configured as https://symfony.com/doc/current/console/hide_commands.html instead?

**Pull request merged**